typings/index.d.ts

```diff
@@ -3616,7 +3616,7 @@ declare module 'discord.js' {
 
   interface PermissionOverwriteOptions extends Partial<Record<PermissionString, boolean | null>> {}
 
-  type PermissionResolvable = BitFieldResolvable<PermissionString, bigint>;
+  type PermissionResolvable = BitFieldResolvable<PermissionString, string, bigint>;
```

Comment:
`BitFieldResolvable` only accepts two type parameters, the second one likely was intended to be `string | bigint`.

That change however would allow `Permissions.resolve("ADD_REAACTIONS")` to compile (and then throw in runtime), which is not wanted behavior.

Since the changes in this PR apply to all `BitField`s, I think we should make a change in `BitFieldResolvable` directly.
To avoid the above loss in typesafety, instead of adding `string` we should add `` `${bigint}` `` there. (Once in the union directly, once in the union of `RecursiveReadonlyArray`.)

Reply:
Okay
I wasn't sure about my typings
I will remove the second argument.
I think that some help for the rest would be necessary, I have not yet grasped all that this implies 🙏

Reply:
Something like that ?
```ts
  type BitFieldResolvable<T extends string, N extends number | bigint> =
    | RecursiveReadonlyArray<T | N | `${bigint}` | Readonly<BitField<T, N>>>
    | T
    | N
    | Readonly<BitField<T, N>>;
```

Reply:
Almost, you also need to add the `${bigint}` part in the outmost union.
```ts
...
| T
| N
| `${bigint}`
...
```